### ISSUE TITLE
Docker images via Jib

### ISSUE THREAD

**Issue by viesti** [juxt/pack.alpha#41]
Currently, I'm putting jars into Docker image, but [Jib](https://github.com/GoogleContainerTools/jib/tree/master/jib-core#adding-jib-core-to-your-build) looks like a tool to make the Docker image building step way faster. Imagine pushing new container to Kubernetes right from the REPL :)

**Comment by SevereOverfl0w** [comment]
Looks awesome, would love to see this. I haven't looked at the API to see, but I would hope that the the internal pack api would make support easy if someone wanted to add it.

**Comment by viesti** [comment]
Initial glance looks like a good fit, let's see if I can shape something up during the holidays :)

**Comment by SevereOverfl0w** [comment]
Something very similar to skinny jar would be perfect for this.  Using an uberjar style for the docker images would be a mistake I think, as you're in an environment where you can do `-cp` easily without it impacting your ease of running the image.

**Comment by viesti** [comment]
A very small spike with only libs (no project content yet) at https://github.com/viesti/pack.alpha/commit/61e8334bad55b6e34bb3dea0ac4f912b1a130096:

```clojure
0% clojure -A:pack-jib mach.pack.alpha.jib -A:xxx --image-name testing:1
Building testing:1
0% docker run --rm -it testing:1
Clojure 1.10.1
user=> (+ 1 1)
2
```

**Comment by SevereOverfl0w** [comment]
That's awesome! Great work.

Fri Jun 28 12:15:28 GMT+01:00 2019 Kimmo Koskinen <notifications@github.com>:

**Comment by viesti** [comment]
Might be a bit hastily done, but decided to open a PR for eyeballing anyway https://github.com/juxt/pack.alpha/pull/42 :) I tried this out with a pet project, managed to create image to local Docker daemon, output a tarball and push to a Google Cloud Registry and run the pushed image.

**Comment by viesti** [comment]
Closing since #42 got merged in 2769a6224bfb938e777906ea311b3daf7d2220f5.
